packages/plugin-avatar/src/avatar.js

```diff
@@ -0,0 +1,123 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {AvatarUrlBatcher} from './avatar-url-batcher';
+import {AvatarUrlStore} from './avatar-url-store';
+import {patterns} from '@ciscospark/common';
+import {SparkPlugin} from '@ciscospark/spark-core';
+import {User} from '@ciscospark/plugin-user';
+import {defaults} from 'lodash';
+
+const Avatar = SparkPlugin.extend({
+  namespace: `Avatar`,
+
+  children: {
+    batcher: AvatarUrlBatcher
+  },
+
+  session: {
+    store: {
+      default() {
+        return new AvatarUrlStore();
+      },
+      type: `any`
+    }
+  },
+
+  /**
+   * @private
+   * Requests an avatar URL from the api
+   * @param {string} uuid
+   * @param {integer} size
+   * @returns {Promise<Object>}
+   */
+  _fetchAvatarUrl(uuid, size) {
+    return this.store.get(uuid, size)
+      .catch(() =>
+        this.batcher.request(Object.assign({}, {uuid, size}))
```

Comment:
I've tended to put the invocation on the same line as the arrow, to make it clearer that the braces were omitted on purpose.